web/source/osk/visualKeyboard.ts

```diff
@@ -679,6 +679,87 @@ namespace com.keyman.osk {
       }
     }
 
+    get width(): number {
+      if(this._width) {
+        return this._width;
+      } else {
+        return this.loadSizeFromCookie().width;
+      }
+    }
+
+    get height(): number {
+      if(this._height) {
+        return this._height;
+      } else {
+        return this.loadSizeFromCookie().height;
+      }
+    }
+
+    protected loadSizeFromCookie(): {width: number, height: number} {
+      let keyman = com.keyman.singleton;
+      let util = keyman.util;
+
+      var c = util.loadCookie('KeymanWeb_OnScreenKeyboard');
+      var newWidth: number, newHeight: number;
+      if(typeof(c) == 'undefined' || c == null) {
+        newWidth = screen.width * 0.3;
+        newHeight = 144; // 16 * 8; 16 is our preferred default font size.
```

Comment:
Previously, the height was calculated as a factor of width. Why do you have a fixed value here now? That seems like it could result in very distorted OSK on some screens?

Reply:
There was no previous `loadHeightFromCookie` method.  The closest thing?

https://github.com/keymanapp/keyman/blob/869899fbacb425adf1e3f24411dfdbcfbb6a0504/web/source/osk/visualKeyboard.ts#L1817-L1832

I simply combined line 1823 with line 1831 in order to preserve existing behavior.